backoff.go

```diff
@@ -0,0 +1,27 @@
+package dsync
+
+import (
+	"math"
+	"time"
+)
+
+// A BackoffFunc determines the backoff strategy for the associated remote call,
+// given a step 'n', the function should return the duration to sleep before
+// performing the next remote call.
+type BackoffFunc func(n int64) time.Duration
+
+// ConstantBackoff generates a simple back-off strategy of a static backoff duration
+func ConstantBackoff(d time.Duration) BackoffFunc {
+	return func(int64) time.Duration {
+		return d
+	}
+}
+
```

Comment:
I want to understand the purpose of these functions that return functions more. Does this add value or are you being clever?